Python/Product/Common/Infrastructure/TaskExtensions.cs

```diff
@@ -147,7 +150,34 @@ static class TaskExtensions {
                     return t.Result;
                 } catch (AggregateException ex) {
                     ex.Handle(e => e is T);
-                    return default(U);
+                    return default;
+                }
+            });
+        }
+
+        /// <summary>
+        /// Silently handles a collection of exception types
+        /// </summary>
+        public static Task SilenceExceptions(this Task task, IEnumerable<Type> exceptionTypes) {
+            return task.ContinueWith(t => {
+                try {
+                    t.Wait();
+                } catch (AggregateException ex) {
+                    ex.Handle(e => exceptionTypes.Any(f => e.GetType().Equals(f)));
+                }
+            });
+        }
+
+        /// <summary>
+        /// Silently handles a collection of exception types
+        /// </summary>
+        public static Task<U> SilenceExceptions<U>(this Task<U> task, IEnumerable<Type> exceptionTypes) {
+            return task.ContinueWith(t => {
+                try {
+                    return t.Result;
+                } catch (AggregateException ex) {
+                    ex.Handle(e => exceptionTypes.Any(f => e.GetType().Equals(f)));
```

Comment:
`f.IsAssignableTo(e)` would reflect the semantics of `catch` more closely and be less brittle, esp. if the library starts throwing a private derived class.

Reply:
I couldn't find `IsAssignableTo` on the type class, but I found `IsAssignableFrom` so that should work in the same way. 😄